detections/endpoint/windows_potential_appdomainmanager_hijack_artifacts_creation.yml

```diff
@@ -0,0 +1,78 @@
+name: Windows Potential AppDomainManager Hijack Artifacts Creation
+id: be19b369-fd0c-42be-ae97-c10b6c01638f
+version: 1
+date: '2025-12-10'
+author: Teoderick Contreras, Splunk
+status: production
+type: Anomaly
+description: The following analytic detects the creation of an .exe file along with its corresponding .exe.config and a .dll in the same directory, which is a common pattern indicative of potential AppDomain hijacking or CLR code injection attempts. This behavior may signal that a malicious actor is attempting to load a rogue assembly into a legitimate application's AppDomain, allowing code execution under the context of a trusted process.
+data_source:
+- Sysmon EventID 11
+search: |
+  | tstats `security_content_summariesonly` count min(_time) AS firstTime max(_time) AS lastTime from datamodel=Endpoint.Filesystem
+  where Filesystem.file_name IN ("*.exe", "*.exe.config", "*.dll") AND Filesystem.file_path IN
+    ("*\\windows\\fonts\\*", "*\\temp\\*", "*\\users\\public\\*", "*\\windows\\debug\\*","*\\Users\\Administrator\\Music\\*", "*\\Windows\\servicing\\*", "*\\Users\\Default\\*", "*Recycle.bin*", "*\\Windows\\Media\\*", "*\\Windows\\repair\\*", "*\\PerfLogs\\*", "*\\programdata\\*")
+  AND Filesystem.action = "created"
+  by Filesystem.action Filesystem.dest Filesystem.file_access_time Filesystem.file_create_time Filesystem.file_hash Filesystem.file_modify_time Filesystem.file_name Filesystem.file_path Filesystem.file_acl Filesystem.file_size Filesystem.process_guid Filesystem.process_id Filesystem.user Filesystem.vendor_product
+  | `drop_dm_object_name("Filesystem")`
+  | stats values(file_name) AS files_found
+  values(user) AS users
+  min(firstTime) AS firstTime max(lastTime) AS lastTime
+  BY dest process_guid
+  | eval exe_files = mvfilter(match(files_found, "\.exe$") AND NOT match(files_found, "\.exe\.config$"))
+  | eval config_files = mvfilter(match(files_found, "\.exe\.config$"))
+  | eval exe_base_names = mvmap(exe_files, replace(exe_files, "\.exe$", ""))
+  | eval config_base_names = mvmap(config_files, replace(config_files, "\.exe\.config$", ""))
+  | eval matching_base_names = mvmap(exe_base_names, if(mvfind(config_base_names, VALUE) != -1, VALUE, null()))
+  | eval file_count = mvcount(files_found)
+  | eval match_count = mvcount(matching_base_names)
+  | where file_count = 3 AND  match_count > 0
```

Comment:
You are counting all files created by a process `values(file_name) AS files_found` and then matching only on 3.

This could lead to a break in logic if the process in question creates more than 3 files at a time.

Is this intentional?